### PR TITLE
update repo references usage after core-retrieval changes

### PR DESCRIPTION
**Pull request by erizocosmico**

### CODE REVIEW

File: storage/database.go [hunk 1]
```diff
@@ -54,45 +55,74 @@ func (s *dbRepoStore) GetByEndpoints(endpoints ...string) ([]*model.Repository,
...
+		var refStore model.ReferenceStore
+		kallax.StoreFrom(&refStore, store)
+
+		refs, err := refStore.FindAll(model.NewReferenceQuery().FindByRepository(repo.ID))
+		if err != nil {
+			return err
+		}
+
+		for _, ref := range refs {
+			if err := refStore.Delete(ref); err != nil {
```
Comment (erizocosmico) [inline storage/database.go]: instead of diffing for changes, which can get hairy, I decided to just delete+create

File: storage/database.go [hunk 2]
```diff
@@ -54,45 +55,74 @@ func (s *dbRepoStore) GetByEndpoints(endpoints ...string) ([]*model.Repository,
...
+			// Some references may come from the database, so they can't be inserted
+			// because they are marked as already persisted. Can't be updated either
+			// because we just deleted them all.
+			var emptyModel kallax.Model
+			ref.Model = emptyModel
+			// Can't use refStore.Insert(ref) because that would trigger an update on
+			// the repository, which causes an error because there are no affected rows
+			// so we have to resort to the generic store Insert, which does not perform
+			// updates/inserts of any relationships.
+			err := store.GenericStore().Insert(model.Schema.Reference.BaseSchema, ref)
```
Comment (erizocosmico) [inline storage/database.go]: we could get rid of this by getting rid of the `Repository` relationship handled via kallax and using a `RepositoryID` field instead in core-retrieval. But having the relationship it's handy for querying as such, so imho, i'd leave this as is even if it's not very pretty

File: storage/database.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ func (s *dbRepoStore) Create(repo *model.Repository) error {
 }
 
 func (s *dbRepoStore) Get(id kallax.ULID) (*model.Repository, error) {
-	q := model.NewRepositoryQuery().FindByID(id)
+	q := model.NewRepositoryQuery().WithReferences(nil).FindByID(id)
```
Comment (ajnavarro) [inline storage/database.go]: Why is needed call explicitly `WithReferences` to nil value?

File: storage/database.go (hunk 1 above)
Comment (ajnavarro) [reply storage/database.go]: But this is not atomic, right? Should be in a transaction.

File: storage/database.go (hunk 3 above)
Comment (erizocosmico) [reply storage/database.go]: Because we're not passing any condition to filter the references, we want them all

File: storage/database.go (hunk 1 above)
Comment (erizocosmico) [reply storage/database.go]: It's in a transaction https://github.com/src-d/borges/pull/210/files/0aa0817bbdde008381ca0d866f4739664b06c2b2#diff-97b2ac718f2c9a1379f7f9d43fb9dc34R92

**Pull request merged**